traefik.go

```diff
@@ -91,8 +94,15 @@ func main() {
 				if err == nil {
 					currentConfiguration = configuration
 					configurationRouter = newConfigurationRouter
-					srv.Stop(time.Duration(globalConfiguration.GraceTimeOut) * time.Second)
-					time.Sleep(3 * time.Second)
+					oldServer := srv
+					newsrv := prepareServer(configurationRouter, globalConfiguration, oldServer, loggerMiddleware, metrics)
+					go startServer(newsrv, globalConfiguration)
+					srv = newsrv
+					time.Sleep(2 * time.Second)
```

Comment:
This feels a little bit *hacky* :sweat_smile:. Why not using go channel, because you have no real guarantees that the goroutines started — although 2 seconds seems long enough but you never know with *racey* thingy..